src/aosm/azext_aosm/definition_folder/reader/artifact_definition.py

```diff
@@ -4,35 +4,60 @@
 # --------------------------------------------------------------------------------------------
 
 import json
+import inspect
 from pathlib import Path
 
 from azext_aosm.definition_folder.reader.base_definition import BaseDefinitionElement
 from azext_aosm.common.artifact import ARTIFACT_TYPE_TO_CLASS, BaseArtifact
 
+from azext_aosm.configuration_models.common_parameters_config import (
+    BaseCommonParametersConfig,
+)
+from azext_aosm.common.command_context import CommandContext
+from knack.log import get_logger
+
 from typing import List
 
+logger = get_logger(__name__)
 
-class ArtifactDefinitionElement(BaseDefinitionElement):
-    """Definition for Artifact"""
 
-    artifacts: List[BaseArtifact]
+class ArtifactDefinitionElement(BaseDefinitionElement):
+    """Definition for Artifact"""  # TODO: Is this actually an artifact manifest?
 
     def __init__(self, path: Path, only_delete_on_clean: bool):
         super().__init__(path, only_delete_on_clean)
+        logger.debug("ArtifactDefinitionElement path: %s", path)
         artifact_list = json.loads((path / "artifacts.json").read_text())
-        self.artifacts = []
-        print(ARTIFACT_TYPE_TO_CLASS)
-        for artifact in artifact_list:
-            if "type" not in artifact or artifact["type"] not in ARTIFACT_TYPE_TO_CLASS:
-                raise ValueError("Artifact type is missing or invalid")
-            self.artifacts.append(
-                ARTIFACT_TYPE_TO_CLASS[artifact["type"]].from_dict(artifact)
-            )
-
-    def deploy(self):
+        self.artifacts = [self.create_artifact_object(artifact) for artifact in artifact_list]
+
+    def create_artifact_object(self, artifact: dict) -> BaseArtifact:
+        """
+        Use the inspect module to identify the artifact class's required fields and
+        create an instance of the class using the supplied artifact dict.
+        """
+        if "type" not in artifact or artifact["type"] not in ARTIFACT_TYPE_TO_CLASS:
+            raise ValueError("Artifact type is missing or invalid")
+        class_sig = inspect.signature(ARTIFACT_TYPE_TO_CLASS[artifact["type"]].__init__)
+        class_args = [arg for arg, _ in class_sig.parameters.items() if arg != 'self']

```

Comment:
Add a couple of comments to explain how we are using this module as it isn't obvious from reading the code.